quail/processes/wps_climdex_su.py

```diff
@@ -0,0 +1,140 @@
+    daily maximum temperature exceeds 25 degreesCelsius
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "build_rda": 90,
+            },
+        )
+        inputs = [
+            ComplexInput(
+                "climdex_input",
+                "climdexInput",
+                abstract="R Object of type climdexInput",
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64")
+                ],
+            ),
+            LiteralInput(
+                "output_path",
+                "Output file name",
+                abstract="Filename to store the count of days where tmax > 25 degC for each year (extension .rda)",
+                data_type="string",
+            ),
+            log_level,
+        ]
+
+        outputs = [
+            ComplexOutput(
+                "summer_days_file",
+                "Summer days output file",
+                abstract="A vector containing the number of summer days for each year",
+                supported_formats=[
+                    Format("application/x-gzip", extension=".rda", encoding="base64")
+                ],
+            ),
+        ]
+
+        super(ClimdexSU, self).__init__(
+            self._handler,
+            identifier="climdex_su",
+            title="Climdex Summer Days",
+            abstract="Computes the annual count of days where daily maximum temperature exceeds 25 degreesCelsius",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def _handler(self, request, response):
+        climdex_input, output_path, loglevel = [
+            arg[0] for arg in collect_args(request, self.workdir).values()
+        ]
+
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+        climdex = get_package("climdex.pcic")
+        base = get_package("base")
+
+        log_handler(
+            self,
+            response,
+            "Processing Summer Days",
+            logger,
+            log_level=loglevel,
+            process_step="process",
+        )
+
+        # First load the climdexInput object into the environment
+        # Then assign that object a name in the python environment
+        ci = robjects.r(robjects.r("load(file='{}')".format(climdex_input))[0])
```

Comment:
I don't feel comfortable using the first object from an `rda` file. I believe there may be many objects residing in a file, and the object that we want might not be the first one. What do you think about adding a string `LiteralInput` that can point out which object in the input file is going to be used. In my repo, I am maintaining a ComplexInput, [ci_file](https://github.com/pacificclimate/quail/compare/i6-frost-days#diff-aacdb9cbf0bda85c9c6ef3d132d0963f35eac820ca7c522d32c10c3ff72c4800R22), and a LteralInput, [ci_name](https://github.com/pacificclimate/quail/compare/i6-frost-days#diff-aacdb9cbf0bda85c9c6ef3d132d0963f35eac820ca7c522d32c10c3ff72c4800R22).

Reply:
I see what you're saying, I was trying to avoid having that extra step, but it might be unavoidable. I also like that you accept .rds and .rda I will add the same

Reply:
Hopefully this makes sense, but with both being `application/x-gzip` I believe birdy does not know how to distinguish between the 2 formats, so just chooses the first one for the input file because of [guess_type](https://github.com/bird-house/birdy/blob/ed4998154fd723a079c1b5bb773c860cd5affcdb/birdy/utils.py#L125). I tried switching them around and that seemed to be the case. It might still work, but it will take some experimenting.